minesweeper.py

```diff
@@ -0,0 +1,196 @@
+from collections import defaultdict
+from random import sample
+
+
+class Square(object):
+
+    def __init__(self):
+        """Creates minesweeper square with mine and clicked attributes."""
+        self.clicked = False
+        self.flagged = False
+        self._mine = False
+
+    def __repr__(self):
+        return "<Sq clicked: {}>".format(self.clicked, self._mine)
+
+
+class Board(object):
+
+    def __init__(self, mine_num=20, height=10, width=10):
+        """Creates a 10x10 board with specified number of mines."""
+        self.board = defaultdict(dict)
```

Comment:
Board could be a 2D array -- no need to have it a dict of dicts. The bounds are known and the indexes are contiguous.